pkg/sign/content.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sign
+
+import (
+	"fmt"
+
+	protobundle "github.com/sigstore/protobuf-specs/gen/pb-go/bundle/v1"
+	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
+	protodsse "github.com/sigstore/protobuf-specs/gen/pb-go/dsse"
+)
+
+type Content interface {
+	// Return the data to be signed
+	Prepare() []byte
```

Comment:
I think we should rename this to something... different... 😄 
As this preparation step is extremely critical to achieve good security, I think this should be more clear in the name. In the DSSE example below, DSSE PAE (Pre Auth Encoding) is used. And I'm thinking if the method name on the interface should be similar? `PAE`, or `Canonicalize`? 
My vote would be PAE as it's commonly used technique to prepare data before computing a digest os signing.

Reply:
I updated this to be `PreAuthEncoding`